src/k8-types/src/core/pod.rs

```diff
@@ -232,6 +234,23 @@ pub struct EmptyDirVolumeSource {
     pub size_limit: Option<String>,
 }
 
+#[derive(Deserialize, Serialize, Default, Debug, Clone, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct CsiVolumeSource {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub driver: Option<String>,
+    pub read_only: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub volume_attributes: Option<CsiVolumeAttributes>,
```

Comment:
I think that we can add also: `fs_type: Option<String>` just in case that in the future we want to use a different value

Reply:
not sure about the other fields, like `volumeHandle`

Reply:
https://kubernetes.io/docs/concepts/storage/volumes/#csi I think is a little bit cleaner a reference. I'll add both fs_type and volume_handle.  They both seem like they might be useful.